diffkemp/simpll/simpll.py

```diff
@@ -75,41 +86,86 @@ def simplify_modules_diff(first, second, fun_first, fun_second, var,
-                                        fun_pair_yaml["second"]]
-                        ]
-
-                        objects_to_compare.append(tuple(fun_pair))
+                if "function-results" in simpll_result:
+                    for fun_result in simpll_result["function-results"]:
+                        res_left = fun_result["first"]
+                        res_right = fun_result["second"]
+                        # Create the vertex and its edges.
+                        vertex = ComparisonGraph.Vertex(
+                            (res_left["function"], res_right["function"]),
+                            Result.Kind.from_string(fun_result["result"]),
+                            (res_left["file"] if "file" in res_left else None,
+                             res_right["file"] if "file" in res_right
+                             else None),
+                            (res_left["line"] if "line" in res_left
+                             else None,
+                             res_right["line"] if "line" in res_right
+                             else None)
+                        )
+                        if "calls" in res_left:
+                            for res, side in [(res_left,
+                                               ComparisonGraph.Side.LEFT),
+                                              (res_right,
+                                               ComparisonGraph.Side.RIGHT)]:
+                                for callee in res["calls"]:
+                                    # Do not insert abstractions.
+                                    if any(map(lambda x: callee["function"].
+                                               startswith(x),
+                                               abstraction_prefixes)):
+                                        continue
+                                    edge = ComparisonGraph.Edge(
+                                        callee["function"],
+                                        callee["file"],
+                                        int(callee["line"]))
+                                    graph.add_edge(vertex, side, edge)
+                        # Add non-function differences.
+                        if "differing-objects" in fun_result:
+                            for nonfun_diff in fun_result["differing-objects"]:
+                                if "body-first" in nonfun_diff:
+                                    # Syntax difference
+                                    diff_obj = ComparisonGraph.SyntaxDiff(
+                                        nonfun_diff["name"],
+                                        nonfun_diff["function"],
+                                        (nonfun_diff["stack-first"],
+                                         nonfun_diff["stack-second"]),
+                                        (nonfun_diff["body-first"],
+                                         nonfun_diff["body-second"])
+                                    )
```

Comment:
Same as above for `Vertex` - the constructor could take `nonfun_diff` as a single parameter, which would simplify this function heavily.